### ISSUE TITLE
FastSync failing to synchronize. Asking to reindex

### ISSUE THREAD

**Issue by sirdavalos** [btcpayserver/btcpayserver#1145]
BTCPay and FastSyncwas working fine on AWS till last week.
I had to move and reinstall to DigitalOcean droplet a few hours ago.

Full node is ok. However, if I use "./load-utxo-set.sh" to enable FastSync then bitcoin daemon failing to synchronize.

**docker logs --tail 10 btcpayserver_bitcoind:**

2019-11-11T17:40:37Z BerkeleyEnvironment::Open: LogDir=/walletdata/mainnet/database ErrorFile=/walletdata/mainnet/db.log

2019-11-11T17:40:37Z [default wallet] nFileVersion = 180000

2019-11-11T17:40:37Z [default wallet] Keys: 2001 plaintext, 0 encrypted, 2001 w/ metadata, 2001 total. Unknown wallet records: 0

2019-11-11T17:40:37Z [default wallet] Wallet completed loading in             168ms

2019-11-11T17:40:37Z Error: Prune: last wallet synchronisation goes beyond pruned data. You need to -reindex (download the whole blockchain again in case of pruned node)
Error: Prune: last wallet synchronisation goes beyond pruned data. You need to -reindex (download the whole blockchain again in case of pruned node)

2019-11-11T17:40:37Z [default wallet] Releasing wallet

2019-11-11T17:40:37Z Shutdown: In progress...

2019-11-11T17:40:37Z scheduler thread interrupt

2019-11-11T17:40:37Z Shutdown: done`

**docker logs generated_nbxplorer_1:**

> info: Configuration:  Data Directory: /datadir
> info: Configuration:  Configuration File: /datadir/Main/settings.config
> info: Configuration:  Network: Mainnet
> info: Configuration:  Supported chains: BTC
> info: Configuration:  Network: Mainnet
> info: Configuration:  Supported chains: BTC
> info: Configuration:  BTC: Testing RPC connection to http://bitcoind:43782/
> info: Configuration:  BTC: Transient error 'Loading block index...', retrying soon...
> Hosting environment: Production
> Content root path: /app/
> Now listening on: http://0.0.0.0:32838
> Application started. Press Ctrl+C to shut down.
> info: Configuration:  BTC: Transient error 'Loading block index...', retrying soon...
> info: Configuration:  BTC: Transient error 'Loading block index...', retrying soon...
> info: Configuration:  BTC: Transient error 'Loading block index...', retrying soon...
> info: Configuration:  BTC: Transient error 'Loading block index...', retrying soon...
> info: Configuration:  BTC: Transient error 'Verifying blocks...', retrying soon...
> fail: Configuration:  BTC: Error connecting to RPC server Response status code does not indicate success: 503 (Service Unavailable).
> info: Configuration:  BTC: Testing again in 5 seconds
> info: Configuration:  BTC: Testing RPC connection to http://bitcoind:43782/
> info: Configuration:  BTC: Transient error 'Loading block index...', retrying soon...
> info: Configuration:  BTC: Transient error 'Rewinding blocks...', retrying soon...
> fail: Configuration:  BTC: Error connecting to RPC server Response status code does not indicate success: 503 (Service Unavailable).
> info: Configuration:  BTC: Testing again in 10 seconds
> info: Configuration:  BTC: Testing RPC connection to http://bitcoind:43782/
> info: Configuration:  BTC: Transient error 'Loading block index...', retrying soon...
> info: Configuration:  BTC: Transient error 'Loading block index...', retrying soon...
> info: Configuration:  BTC: Transient error 'Loading block index...', retrying soon...
> fail: Configuration:  BTC: Error connecting to RPC server Response status code does not indicate success: 503 (Service Unavailable).
> info: Configuration:  BTC: Testing again in 15 seconds
> info: Configuration:  BTC: Testing RPC connection to http://bitcoind:43782/
> info: Configuration:  BTC: Transient error 'Rewinding blocks...', retrying soon...
> info: Configuration:  BTC: Transient error 'Verifying blocks...', retrying soon...
> fail: Configuration:  BTC: Error connecting to RPC server Response status code does not indicate success: 503 (Service Unavailable).
> info: Configuration:  BTC: Testing again in 20 seconds
> info: Configuration:  BTC: Testing RPC connection to http://bitcoind:43782/
> info: Configuration:  BTC: Transient error 'Loading block index...', retrying soon...
> info: Configuration:  BTC: Transient error 'Loading block index...', retrying soon...
> info: Configuration:  BTC: Transient error 'Rewinding blocks...', retrying soon...
> fail: Configuration:  BTC: Error connecting to RPC server Response status code does not indicate success: 503 (Service Unavailable).
> info: Configuration:  BTC: Testing again in 25 seconds

**Comment by britttttk** [comment]
Since you're trying to fast sync, I think you can just `rm -r blocks`  in `_data` and try to run `/.load-utxo-set.sh` again.

**Comment by sirdavalos** [comment]
**Removed both data and chainstate. Not working**

> rm -rf /var/lib/docker/volumes/generated_bitcoin_datadir/_data/blocks

> rm -rf /var/lib/docker/volumes/generated_bitcoin_datadir/_data/chainstate



> 2019-11-12T03:49:15Z BerkeleyEnvironment::Open: LogDir=/walletdata/mainnet/database ErrorFile=/walletdata/mainnet/db.log
> 2019-11-12T03:49:15Z [default wallet] nFileVersion = 180000
> 2019-11-12T03:49:15Z [default wallet] Keys: 2001 plaintext, 0 encrypted, 2001 w/ metadata, 2001 total. Unknown wallet records: 0
> 2019-11-12T03:49:15Z [default wallet] Wallet completed loading in              44ms
> 2019-11-12T03:49:15Z Error: Prune: last wallet synchronisation goes beyond pruned data. You need to -reindex (download the whole blockchain again in case of pruned node)
> Error: Prune: last wallet synchronisation goes beyond pruned data. You need to -reindex (download the whole blockchain again in case of pruned node)
> 2019-11-12T03:49:15Z [default wallet] Releasing wallet
> 2019-11-12T03:49:15Z Shutdown: In progress...
> 2019-11-12T03:49:15Z scheduler thread interrupt
> 2019-11-12T03:49:15Z Shutdown: done

**Comment by britttttk** [comment]
That's weird, can you make sure you have pruning still enabled? 
```
cd /root/btcpayserver-docker
echo $BTCPAYGEN_ADDITIONAL_FRAGMENTS
```

**Comment by sirdavalos** [comment]
> root@ubuntu-btcpay:~/BTCPayServer/btcpayserver-docker/contrib/FastSync# echo $BTCPAYGEN_ADDITIONAL_FRAGMENTS
> opt-save-storage-xs;opt-save-memory

**Comment by andrepadez** [comment]
Also having the same trouble...
```$BTCPAYGEN_ADDITIONAL_FRAGMENTS
opt-save-storage-xs```

**Comment by pavlenex** [comment]
Ping @NicolasDorier at least 3 people reported having issues with FastSync.

**Comment by pavlenex** [comment]
Def something strange going on there. After the fast sync, the progress bar seems to be stuck at 
0% for me.

![Screenshot 2019-11-12 at 13 19 08](https://user-images.githubusercontent.com/36959754/68671196-086ae580-054f-11ea-8fed-ef8e9820200e.png)

## NBX
```
info: Configuration:  Data Directory: /datadir
info: Configuration:  Configuration File: /datadir/Main/settings.config
info: Configuration:  Network: Mainnet
info: Configuration:  Supported chains: BTC
info: Configuration:  Network: Mainnet
info: Configuration:  Supported chains: BTC
info: Configuration:  BTC: Testing RPC connection to http://bitcoind:43782/
info: Configuration:  BTC: Transient error 'Loading block index...', retrying soon...
Hosting environment: Production
Content root path: /app/
Now listening on: http://0.0.0.0:32838
Application started. Press Ctrl+C to shut down.
info: Configuration:  BTC: Transient error 'Loading block index...', retrying soon...
info: Configuration:  BTC: Transient error 'Loading block index...', retrying soon...
info: Configuration:  BTC: Transient error 'Loading block index...', retrying soon...
info: Configuration:  BTC: Transient error 'Loading block index...', retrying soon...
info: Configuration:  BTC: RPC connection successful
info: Configuration:  BTC: Full node version detected: 180000
info: Configuration:  BTC: Node banlist loaded
info: Events:         BTC: Node state changed: NotStarted => CoreSynching
```
Applied `btcpay-restart.sh`, it started syncing from scratch it seems

![Screenshot 2019-11-12 at 13 19 08](https://user-images.githubusercontent.com/36959754/68671317-5e3f8d80-054f-11ea-84aa-e31a374afe5a.png)

## Bitcoind

```
2019-11-12T12:09:41Z Bitcoin Core version v0.18.0 (release build)
2019-11-12T12:09:41Z Assuming ancestors of block 0000000000000000000f1c54590ee18d15ec70e68c8cd4cfbadb1b4f11697eee have valid signatures.
2019-11-12T12:09:41Z Setting nMinimumChainWork=0000000000000000000000000000000000000000051dc8b82f450202ecb3d471
2019-11-12T12:09:41Z Prune configured to target 25000 MiB on disk for block and undo files.
2019-11-12T12:09:41Z Using the 'sse4(1way),sse41(4way)' SHA256 implementation
2019-11-12T12:09:41Z Default data directory /home/bitcoin/.bitcoin
2019-11-12T12:09:41Z Using data directory /home/bitcoin/.bitcoin
2019-11-12T12:09:41Z Config file: /home/bitcoin/.bitcoin/bitcoin.conf
2019-11-12T12:09:41Z Using at most 125 automatic connections (1024 file descriptors available)
2019-11-12T12:09:41Z Using 16 MiB out of 32/2 requested for signature cache, able to store 524288 elements
2019-11-12T12:09:41Z Using 16 MiB out of 32/2 requested for script execution cache, able to store 524288 elements
2019-11-12T12:09:41Z Using 2 threads for script verification
2019-11-12T12:09:41Z scheduler thread start
2019-11-12T12:09:41Z WARNING: the RPC server is not safe to expose to untrusted networks such as the public internet
2019-11-12T12:09:41Z HTTP: creating work queue of depth 16
2019-11-12T12:09:41Z No rpcpassword set - using random cookie authentication.
2019-11-12T12:09:41Z Generated RPC authentication cookie /home/bitcoin/.bitcoin/.cookie
2019-11-12T12:09:41Z Using rpcauth authentication.
2019-11-12T12:09:41Z HTTP: starting 4 worker threads
2019-11-12T12:09:41Z Using wallet directory /walletdata/mainnet
2019-11-12T12:09:41Z init message: Verifying wallet(s)...
2019-11-12T12:09:41Z Using BerkeleyDB version Berkeley DB 4.8.30: (April  9, 2010)
2019-11-12T12:09:41Z Using wallet /walletdata/mainnet
2019-11-12T12:09:41Z BerkeleyEnvironment::Open: LogDir=/walletdata/mainnet/database ErrorFile=/walletdata/mainnet/db.log
2019-11-12T12:09:41Z init message: Loading banlist...
2019-11-12T12:09:41Z Cache configuration:
2019-11-12T12:09:41Z * Using 2.0 MiB for block index database
2019-11-12T12:09:41Z * Using 8.0 MiB for chain state database
2019-11-12T12:09:41Z * Using 440.0 MiB for in-memory UTXO set (plus up to 286.1 MiB of unused mempool space)
2019-11-12T12:09:41Z init message: Loading block index...
2019-11-12T12:09:41Z Opening LevelDB in /home/bitcoin/.bitcoin/blocks/index
2019-11-12T12:09:48Z Opened LevelDB successfully
2019-11-12T12:09:48Z Using obfuscation key for /home/bitcoin/.bitcoin/blocks/index: 0000000000000000
2019-11-12T12:09:55Z LoadBlockIndexDB: last block file = 0
2019-11-12T12:09:55Z LoadBlockIndexDB: last block file info: CBlockFileInfo(blocks=1, size=293, heights=0...0, time=2009-01-03...2009-01-03)
2019-11-12T12:09:55Z Checking all blk files are present...
2019-11-12T12:09:55Z Opening LevelDB in /home/bitcoin/.bitcoin/chainstate
2019-11-12T12:09:55Z Opened LevelDB successfully
2019-11-12T12:09:55Z Using obfuscation key for /home/bitcoin/.bitcoin/chainstate: 78a6e0b7fa0c336d
2019-11-12T12:09:55Z Loaded best chain: hashBestChain=000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f height=0 date=2009-01-03T18:15:05Z progress=0.000000
2019-11-12T12:09:55Z init message: Rewinding blocks...
2019-11-12T12:09:56Z init message: Verifying blocks...
2019-11-12T12:09:56Z  block index           14938ms
2019-11-12T12:09:56Z init message: Loading wallet...
2019-11-12T12:09:56Z BerkeleyEnvironment::Open: LogDir=/walletdata/mainnet/database ErrorFile=/walletdata/mainnet/db.log
2019-11-12T12:09:56Z [default wallet] nFileVersion = 180000
2019-11-12T12:09:56Z [default wallet] Keys: 2001 plaintext, 0 encrypted, 2001 w/ metadata, 2001 total. Unknown wallet records: 0
2019-11-12T12:09:56Z [default wallet] Wallet completed loading in             120ms
2019-11-12T12:09:56Z [default wallet] setKeyPool.size() = 2000
2019-11-12T12:09:56Z [default wallet] mapWallet.size() = 0
2019-11-12T12:09:56Z [default wallet] mapAddressBook.size() = 0
2019-11-12T12:09:56Z Unsetting NODE_NETWORK on prune mode
2019-11-12T12:09:56Z init message: Pruning blockstore...
2019-11-12T12:09:56Z mapBlockIndex.size() = 540001
2019-11-12T12:09:56Z nBestHeight = 0
2019-11-12T12:09:56Z Imported mempool transactions from disk: 0 succeeded, 0 failed, 0 expired, 0 already there
2019-11-12T12:09:56Z torcontrol thread start
2019-11-12T12:09:56Z Bound to [::]:39388
2019-11-12T12:09:56Z Bound to 0.0.0.0:39388
2019-11-12T12:09:56Z init message: Loading P2P addresses...
2019-11-12T12:09:56Z Loaded 5160 addresses from peers.dat  62ms
2019-11-12T12:09:56Z init message: Starting network threads...
2019-11-12T12:09:56Z dnsseed thread start
2019-11-12T12:09:56Z init message: Done loading
2019-11-12T12:09:56Z opencon thread start
2019-11-12T12:09:56Z addcon thread start
2019-11-12T12:09:56Z net thread start
2019-11-12T12:09:56Z msghand thread start
2019-11-12T12:09:59Z New outbound peer connected: version: 70015, blocks=603458, peer=1
2019-11-12T12:10:03Z Pre-allocating up to position 0x100000 in rev00000.dat
```

**Comment by pavlenex** [comment]
But then look at this, bitcoind uses CPU like crazy.

`btcpayserver_bitcoind               123.98%`

```

CONTAINER ID        NAME                                CPU %               MEM USAGE / LIMIT     MEM %               NET I/O             BLOCK I/O           PIDS
0656e6b404ce        letsencrypt-nginx-proxy-companion   0.18%               7.508MiB / 3.844GiB   0.19%               1.43kB / 0B         13.1MB / 4.1kB      12
2f4c2b30fab5        generated_btcpayserver_1            0.19%               180.4MiB / 3.844GiB   4.58%               2.35MB / 10.9MB     5.27MB / 4.1kB      27
4819120e666b        nginx                               0.00%               2.984MiB / 3.844GiB   0.08%               11MB / 11.2MB       7.29MB / 0B         2
d8bbec1b5d3a        btcpayserver_lnd_bitcoin            0.13%               1.285MiB / 3.844GiB   0.03%               1.39kB / 0B         2.11MB / 4.1kB      3
59bb5c9e52ea        btcpayserver_bitcoind               123.98%             595.6MiB / 3.844GiB   15.13%              2.72GB / 45.8MB     8.9MB / 2.41GB      16
dcad686e62e2        tor-gen                             0.20%               4.934MiB / 3.844GiB   0.13%               1.43kB / 42B        7.2MB / 4.1kB       10
f14c1a54e1f4        generated_lnd_bitcoin_rtl_1         0.00%               29.4MiB / 3.844GiB    0.75%               1.39kB / 0B         6.03MB / 0B         8
cb8f0f70f3a0        generated_nbxplorer_1               0.20%               78.58MiB / 3.844GiB   2.00%               443kB / 426kB       9.39MB / 8.19kB     17
f8d292ddcff1        tor                                 0.03%               22.12MiB / 3.844GiB   0.56%               410kB / 1.32MB      13.8MB / 69.6kB     1
48a7e6a8529f        generated_postgres_1                0.00%               9.453MiB / 3.844GiB   0.24%               17.6kB / 37.8kB     20.5MB / 594kB      8
```

**Comment by pavlenex** [comment]
I'm an idiot, it seems my issue is unrelated.

**Comment by NicolasDorier** [comment]
Fuck, this seems to be since I added the wallet feature back to Bitcoin Core for FullyNoded which need it....

```
Error: Prune: last wallet synchronisation goes beyond pruned data. You need to -reindex (download the whole blockchain again in case of pruned node)
Error: Prune: last wallet synchronisation goes beyond pruned data. You need to -reindex (download the whole blockchain again in case of pruned node)
```

**Comment by NicolasDorier** [comment]
There is no good solution to this, if you don't use bitcoin core wallet, delete it:

```bash
rm -rf /var/lib/docker/volumes/generated_bitcoin_wallet_datadir/_data/
mkdir /var/lib/docker/volumes/generated_bitcoin_wallet_datadir/_data/
```

Shit we need to document that.

**Comment by NicolasDorier** [comment]
How can we make this more discoverable?
The problem is that the first boot of bitcoind is generating the wallet. Then when you stop bitcoind, put fastsync and restart, it sees tthat the wallet was created in the early blocks and spout out this message.

**Comment by pavlenex** [comment]
Is there a way to include that into a script? But then it can fuck things up if the wallet is used right?

This is bad, if there's no other way we must remove that,

**Comment by NicolasDorier** [comment]
I can show a warning maybe?

**Comment by NicolasDorier** [comment]
I will show a warning.

**Comment by pavlenex** [comment]
We'd have to document it [here](https://docs.btcpayserver.org/faq-and-common-issues/faq-synchronization#can-i-skip-the-synchronization), in fastsync and add a cli warring I guess. Damn.

**Comment by NicolasDorier** [comment]
I just added proper warnings on the cli

https://github.com/btcpayserver/btcpayserver-docker/commit/18ef1d2d462feccf32044bd69d81f732a495c063

**Comment by NicolasDorier** [comment]
> Since you're trying to fast sync, I think you can just rm -r blocks in _data and try to run ./load-utxo-set.sh again.

@britttttk the download hash is checked, so normally this don't happen (except if there is corruption while you are untar-ing, but this should not happen)

**Comment by britttttk** [comment]
Ooh it’s the wallet, ok I see @NicolasDorier  Will update fastsync doc.

**Comment by pavlenex** [comment]
Closing this as it's been resolved in [18ef1d2](https://github.com/btcpayserver/btcpayserver-docker/commit/18ef1d2d462feccf32044bd69d81f732a495c063)
